src/main/kotlin/gg/skytils/skytilsmod/features/impl/misc/ItemFeatures.kt

```diff
@@ -646,12 +646,11 @@ object ItemFeatures {
                     }
             }
         }
-        if (Skytils.config.showPetCandies && item.item === Items.skull) { // TODO: Use NBT
-            lore?.forEach { line ->
-                candyPattern.find(line)?.let {
-                    stackTip = it.groups[1]!!.value
-                    return@forEach
-                }
+        if (Skytils.config.showPetCandies && item.item === Items.skull) {
+            val petInfoString = getExtraAttributes(item)?.getString("petInfo")
+            if (!petInfoString.isNullOrBlank()) {
+                val petInfo = json.decodeFromString<PetInfo>(petInfoString)
```

Comment:
Is this tested and does it always work? Try just going threw a few pages on ah and seeing if anything breaks.

Reply:
It worked perfect in my testing, no errors occurred.

Reply:
Okay so, it always works, but it still shows the candy used on maxed out pets. Is that fine?

Reply:
Same with 0/10 Candy used, should I let it show 0 or hide it?

Reply:
> Okay so, it always works, but it still shows the candy used on maxed out pets. Is that fine?

What do you mean by maxed out pets? Level 100?


> Same with 0/10 Candy used, should I let it show 0 or hide it?

Do whatever the old code does.

Reply:
From what I understand level 100 pets that can't be upgraded anymore don't show the Candy used line

Reply:
![image](https://github.com/Skytils/SkytilsMod/assets/74718793/6bf0d63d-a465-463b-9b50-c5a11b4a9cd4)

Reply:
Ah, yeah we might want to make it also hide on maxed pets since it would make for a more continuous experience.